universal_transfer_operator/src/universal_transfer_operator/data_providers/base.py

```diff
@@ -1,16 +1,20 @@
 from __future__ import annotations
 
 from abc import ABC
+from typing import Any, Generic, TypeVar
 
 import attr
 from airflow.hooks.base import BaseHook
 
 from universal_transfer_operator.constants import Location
-from universal_transfer_operator.datasets.base import Dataset
+from universal_transfer_operator.datasets.file.base import File
+from universal_transfer_operator.datasets.table import Table
 from universal_transfer_operator.utils import TransferParameters, get_dataset_connection_type
 
+T = TypeVar("T", File, Table)
```

Comment:
Why was this necessary? Perhaps we could give it a more meaningful name (e.g. `DatasetType` or something similar

Reply:
No reason, I'll change it.

Reply:
Done.